code/modules/projectiles/guns/projectile/CMturret.dm

```diff
@@ -0,0 +1,211 @@
+/mob/living/carbon/human
+	var/obj/structure/machinegun/mounted
+
+/mob/living/carbon/human/ClickOn(atom/A, params)
+    if(mounted && Adjacent(mounted))
+        if(next_move < world.time)
+            mounted.shoot(get_turf(A))
+    else
+        . = ..()
+
+
+/obj/structure/machinegun
+	name = "machine gun"
+	desc = "A stationary machine gun."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "mgun+barrier"
+	var/fire_sound = 'sound/weapons/Gunshot_smg.ogg'
+	var/empty_sound = 'sound/weapons/empty.ogg'
+	var/ammo_type = "/obj/item/projectile/bullet/midbullet2"
+	var/ammo = 500
+	var/ammomax = 500
+	var/list/row1 = list()
+	var/list/row2 = list()
+	var/list/row3 = list()
+	var/mob/living/carbon/human/user
+	var/FIRETIME = 1 //tenths of seconds
+	density = 1
+	anchored = 0
+	flags = ON_BORDER
+	var/health = 500
+
+/obj/structure/machinegun/process()
```

Comment:
Well, it doesn't even have to be buckling, it's more just...why is it based on `process`. It would be better if it was based on the current user of the machine rather than a process loop---then you could do check for adjacency (under `check_eye`) and such for whether `unset_machine` should be called or not.